src/generators/schema-generator.php

```diff
@@ -198,6 +199,68 @@ protected function add_schema_blocks_graph_pieces( $graph, $context ) {
 		return $graph;
 	}
 
+	/**
+	 * Finalizes the schema graph after all filtering is done.
+	 *
+	 * @param array             $graph   The current schema graph.
+	 * @param Meta_Tags_Context $context The meta tags context.
+	 *
+	 * @return array The schema graph.
+	 */
+	protected function finalize_graph( $graph, $context ) {
+		$graph = $this->remove_empty_breadcrumb( $graph, $context );
+
+		return $graph;
+	}
+
+	/**
+	 * Removes the breadcrumb schema if empty.
+	 *
+	 * @param array             $graph   The current schema graph.
+	 * @param Meta_Tags_Context $context The meta tags context.
+	 *
+	 * @return array The schema graph with empty breadcrumpbs taken out.
+	 */
+	protected function remove_empty_breadcrumb( $graph, $context ) {
+		if ( $this->helpers->current_page->is_home_static_page() || $this->helpers->current_page->is_home_posts_page() ) {
+			return $graph;
+		}
+
+		// Remove the breadcrumb piece, if it's empty.
+		$index_to_remove = 0;
+		foreach ( $graph as $key => $piece ) {
+			if ( \in_array( 'BreadcrumbList', $this->get_type_from_piece( $piece ), true ) ) {
+				if ( isset( $piece['itemListElement'] ) && is_array( $piece['itemListElement'] ) && count( $piece['itemListElement'] ) === 1 ) {
+					$index_to_remove = $key;
+					break;
+				}
+			}
+		}
+
+		// If the breadcrumb piece has been removed, we should remove its reference from the WebPage node.
+		if ( $index_to_remove !== 0 ) {
+			\array_splice( $graph, $index_to_remove, 1 );
+
+			// Get the type of the WebPage node.
+			$webpage_type = $context->schema_page_type;
```

Comment:
I would maybe change this to $webpage_types and than just remove the else. so if not an array do nothing because it is an array and you can work with that.

Reply:
I wanted to avoid transforming a string variable into an array in the if, so:
`$webpage_types = [ $webpage_types ];`
for consistency purposes.

But I can be convinced that it's not a good enough reason for this addition of complexity 😅

Reply:
nice workaround suggested: https://github.com/Yoast/wordpress-seo/pull/18974/commits/ff3690287a03bbbb9af07f3a69fcab82decbfa24